src/error.rs

```diff
@@ -30,6 +30,12 @@ pub(crate) enum Error {
   FileSearch { source: ignore::Error },
   #[snafu(display("Invalid glob: {}", source))]
   GlobParse { source: globset::Error },
+  #[snafu(display("Failed to parse host port"))]
+  HostPort {
+    source: HostPortParseError,
+  },
+  #[snafu(display("io error {}", source))]
```

Comment:
Yeah, this catchall is also grabbing the `write_all()` calls on cursors. Happy to change them to below?
 
    msg.write_all(&buf).context(MessageSerialize { src: io::Error })?;

Reply:
Sorry, I missed the cursor writes. I believe `Cursor::write*` functions are infallible, since it's backed by a vector. Can those be changed to calls to `Vec::extend_from_slice`, so avoid needing to use `?` or `.unwrap`?